instrumentation/servlet/servlet-3.0/src/main/java/io/opentelemetry/instrumentation/auto/servlet/v3_0/CountingHttpServletResponse.java

```diff
@@ -110,6 +113,30 @@ public void flush() throws IOException {
     public void close() throws IOException {
       delegate.close();
     }
+
+    // New abstract methods introduced in servlet-api 3.1, 3.0 does not have them.
+    // These methods purposefully use reflection to call the delegate object so that they compile
+    // against both 3.0 and 3.1+ servlet-api.
+
+    // @Override
+    public boolean isReady() {
+      try {
+        Method isReady = ServletOutputStream.class.getMethod("isReady");
```

Comment:
This certainly should be cached, no need to lookup on every method call. Also, probably `MethodHandle` will work faster than `Method`

Reply:
Very good point, done. Thanks!